src/server/store/ActivityStore.ts

```diff
@@ -22,13 +23,13 @@ export class ActivityStore {
     }
     const key = this.urlToKey(activity.id)
     await this.db.put(key, activity)
-    await this.addToIndex(getPublished(activity.published), key)
+    await this.addToIndex(activity)
   }
 
   async remove (url: string): Promise<void> {
     const key = this.urlToKey(url)
     await this.db.del(key)
-    await this.removeFromIndex(key)
+    await this.removeFromIndex(url)
```

Comment:
We should remove from the index before the del, else it won't be able to fetch the doc.

Mind adding a test for this? With a fresh store: Add a doc, remove the doc by url, test that there are no index keys and no docs.

Reply:
Fixed tests, modified list test to check ordering.